### PR TITLE
AllowedProcMountTypes - Update DropDisabled[Alpha]Fields behaviour

### PR DESCRIPTION
**Pull request by mason1kwok**
/kind cleanup

Which issue(s) this PR fixes:

ref: #72169

Special notes for your reviewer:

Does this PR introduce a user-facing change?:
```release-note
NONE
```

### CODE REVIEW

File: pkg/api/podsecuritypolicy/util.go [hunk 1]
```diff
@@ -38,3 +35,16 @@ func DropDisabledFields(pspSpec, oldPSPSpec *policy.PodSecurityPolicySpec) {
 		}
 	}
 }
+
+func allowedProcMountTypesInUse(oldPSPSpec *policy.PodSecurityPolicySpec) bool {
```
Comment (mourya007) [inline pkg/api/podsecuritypolicy/util.go]: Can we use AllowedProcMountTypes as the name of the function to relate it with the  function of util_test.go?

Comment (liggitt) [reply pkg/api/podsecuritypolicy/util.go]: this is fine as is, the field is AllowedProcMountTypes, this function is checking if it is in use, and we want it private to the package

Comment (mason1kwok) [reply pkg/api/podsecuritypolicy/util.go]: Thanks for your explanation, it makes me clearer. I saw it from another PR.

**Pull request merged**